### ISSUE TITLE
Backpack v5 compability

### ISSUE THREAD

**Issue by howest-ward** [ziming/json-field-for-backpack#6]
Is it possible to make this awesome field compatible with v5 of backpack?

**Comment by ziming** [comment]
Your wish has been granted.

**Comment by howest-ward** [comment]
Sweet! Thank you!

**Comment by ziming** [comment]
Very silly post from me

Hmm I have a few reasons for choosing >=. One of which is that this is not a customer facing package but an internal staff facing 1 where issues are more acceptable. So by doing >=, at least people don't have to wait for me when future major versions of backpack is out to upgrade (Tho they can still learn how to use a fork or composer patches. I know how but even then I find myself being lazy & end up waiting for the maintainers which sometimes take months/forever until I gave up & fork/composer patch it).

& if it breaks. They can then comment out my admin field & post an issue here on what they are facing in the new major version.

The other reason tho, is I originally only plan to release patches to this package once every year on the same day for personal reasons (just to keep up to date with the latest stable version of the jsoneditor javascript library). So I was hoping >= will reduce the times I need to patch it & make a new release

Hope you give me some time to think about it. If I do release a 1.1.2 in the coming days, I may not delete 1.1.1 though. I once remember a maintainer did something similar & people are not happy about it on twitter as it broke their deployments (I guess the timing of their deployment is when their composer.lock is at "1.1.1" but the maintainer deletes it). But my packages aren't popular enough to cause that kind of issue to people heh.

**Comment by tabacitu** [comment]
Sure - I understand your position / pondering. Take your time. But also take into consideration that:
- [Composer itself heavily recommends having an upper bounds constraint](https://getcomposer.org/doc/faqs/why-are-unbound-version-constraints-a-bad-idea.md);
- the longer you wait, the more potentially "damaging" it will be to delete the 1.1.1 tag, because it'll have more downloads;
- if you don't delete the 1.1.1 tag, you don't really solve anything; even when this package gets to version 9.9.9, you'll still have version 1.1.1 in your history, that can be installed on "any" backpack version, ever; that's the problem I was highlighting;
- I've deleted tags before, when I made mistakes, in packages that had a lot more downloads; it's not that big of a deal 😅 some people get a bad experience... so that MOST PEOPLE do not; the trick is to do it fast 😀

For context, I'm not a "strict" developer, by any means. For example, I like form validations more lax, rather than more strict. I rarely use return types. My methods default to "public". But when it comes to package versions - my opinion is firm, based on experience... "_stricter is better_". "Packages should only be installable on configurations you _know_ will work." 

This is why working with Composer & Packagist is great, and working with NPM can be awful sometimes - people in the JS ecosystem allow their packages to be installed on whatever and let the dev "see what happens". It amounts to an awful developer experience, if you ask me. Much worse than trying to install, telling you it doesn't work, then having to submit a PR to have it work (or making a fork). And by consequence... helping the maintainer maintain the project.

My 2 cents, of course 😀
